p2p/server/server.go

```diff
@@ -147,26 +228,51 @@ func (s *Server) streamHandler(stream network.Stream) {
 	}
 }
 
+func (s *Server) trackClientLatency(start time.Time, err error) {
+	if s.metrics != nil {
+		if err != nil {
+			s.metrics.clientLatencyFailure.Observe(time.Since(start).Seconds())
+		} else {
+			s.metrics.clientLatency.Observe(time.Since(start).Seconds())
+		}
+	}
+}
+
 // Request sends a binary request to the peer. Request is executed in the background, one of the callbacks
 // is guaranteed to be called on success/error.
-func (s *Server) Request(ctx context.Context, pid peer.ID, req []byte, resp func([]byte), failure func(error)) error {
+func (s *Server) Request(
+	ctx context.Context,
+	pid peer.ID,
+	req []byte,
+	resp func([]byte),
+	failure func(error),
+) error {
+	start := time.Now()
 	if len(req) > s.requestLimit {
 		return fmt.Errorf("request length (%d) is longer than limit %d", len(req), s.requestLimit)
 	}
 	if s.h.Network().Connectedness(pid) != network.Connected {
 		return fmt.Errorf("%w: %s", ErrNotConnected, pid)
 	}
 	go func() {
```

Comment:
Should this also be part of an `errgroup.Group` and be awaited?

Reply:
maybe but it is not related to the change